src/Microsoft.AspNetCore.OData/Extensions/ODataServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Adapters;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNet.OData.Query;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Extensions
+{
+    /// <summary>
+    /// Provides extension methods to add odata services.
+    /// </summary>
+    public static class ODataServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds essential OData services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <returns>An <see cref="IODataBuilder"/> that can be used to further configure the OData services.</returns>
+        public static IODataBuilder AddOData(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw Error.ArgumentNull(nameof(services));
+            }
+
+            // Setup per-route dependency injection. When routes are added, additional
+            // per-route classes will be injected, such as IEdmModel and IODataRoutingConventions.
+            services.AddSingleton<IPerRouteContainer, PerRouteContainer>();
+
+            // Add OData options.
+            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<ODataOptions>, ODataOptionsSetup>());
+            services.TryAddEnumerable(ServiceDescriptor.Transient<IConfigureOptions<DefaultQuerySettings>, DefaultQuerySettingsSetup>());
+
+            // Configure MvcCore to use formatters.
+            services.AddMvcCore(options =>
+            {
+                // Add OData input formatters at index 0, which overrides the built-in json and xml formatters.
+                foreach (ODataInputFormatter inputFormatter in ODataInputFormatterFactory.Create())
+                {
+                    options.InputFormatters.Insert(0, inputFormatter);
+                }
+
+                //options.FormatterMappings.SetMediaTypeMappingForFormat("pdf", "application/pdf");
```

Comment:
Leftover commented code or a todo for a future PR?

Reply:
Also, extra empty line below.

Reply:
Yep, I'll remove for now.